core/commands/storage/upload/upload/dc_repair_router.go

```diff
@@ -0,0 +1,265 @@
+			return err
+		}
+		err = emptyCheck(fileHash, lostShardHashes, repairId, repairContractId, fileSize, downloadRewardAmount, repairRewardAmount)
+		if err != nil {
+			return err
+		}
+		ctxParams, err := uh.ExtractContextParams(req, env)
+		if err != nil {
+			return err
+		}
+		ns, err := helper.GetHostStorageConfig(ctxParams.Ctx, ctxParams.N)
+		if err != nil {
+			return err
+		}
+		var price uint64
+		if ns.RepairCustomizedPricing {
+			price = ns.RepairPriceCustomized
+		} else {
+			price = ns.RepairPriceDefault
+		}
+		if price > uint64(repairRewardAmount) {
+			return fmt.Errorf("host desired price %d is greater than request price %d", price, repairRewardAmount)
+		}
+		submitSignedRepairContract(ctxParams, &RepairContractParams{
+			FileHash:             fileHash,
+			FileSize:             fileSize,
+			RepairPid:            repairId,
+			RepairSignTime:       time.Now(),
+			LostShardHashes:      lostShardHashes,
+			RepairContractId:     repairContractId,
+			DownloadContractId:   downloadContractId,
+			RepairRewardAmount:   repairRewardAmount,
+			DownloadRewardAmount: downloadRewardAmount,
+		})
+		return nil
+	},
+}
+
+func submitSignedRepairContract(ctxParams *uh.ContextParams, params *RepairContractParams) {
+	go func() {
+		err := func() error {
+			repaireReq := &guardpb.RepairContract{
+				FileHash:             params.FileHash,
+				FileSize:             params.FileSize,
+				RepairPid:            params.RepairPid,
+				LostShardHash:        params.LostShardHashes,
+				RepairSignTime:       time.Now().UTC(),
+				RepairContractId:     params.RepairContractId,
+				DownloadContractId:   params.DownloadContractId,
+				RepairRewardAmount:   params.RepairRewardAmount,
+				DownloadRewardAmount: params.DownloadRewardAmount,
+			}
+			sig, err := crypto.Sign(ctxParams.N.PrivateKey, repaireReq)
+			if err != nil {
+				return err
+			}
+			repaireReq.RepairSignature = sig
+			err = grpc.GuardClient(ctxParams.Cfg.Services.GuardDomain).WithContext(ctxParams.Ctx, func(ctx context.Context,
+				client guardpb.GuardServiceClient) error {
+				_, err = client.SubmitRepairContract(ctx, repaireReq)
+				if err != nil {
+					return err
+				}
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+			return nil
+		}()
+		if err != nil {
+			log.Errorf("repairer signed and submitted contract error: %s", err.Error())
+		}
+	}()
+}
+
+func emptyCheck(fileHash string, lostShardHashes []string, RepairPid string, repairContractId string, fileSize int64, DownloadRewardAmount int64, RepairRewardAmount int64) error {
+	if fileHash == "" {
+		return fmt.Errorf("file Hash is empty")
+	}
+	if RepairPid == "" {
+		return fmt.Errorf("repair id is empty")
+	}
+	if repairContractId == "" {
+		return fmt.Errorf("repair contract id is empty")
+	}
+	if fileSize == 0 {
+		return fmt.Errorf("file size is 0")
+	}
+	if len(lostShardHashes) == 0 {
+		return fmt.Errorf("lost shard hashes do not specify")
+	}
+	if DownloadRewardAmount == 0 {
```

Comment:
Not sure, seems we need not guarantee the two amounts is bigger than 0, as node should compare the amount with its price, we need not do it.  Maybe in the future, the node will pay us to do  the repair job(amount < 0) for some reason, we need not limit them now.

Reply:
updated